### ISSUE TITLE
Issues with BeJsonEquivalentTo and long json strings

### ISSUE THREAD

**Issue by Numpsy** [cmeeren/Faqt#6]
Hi,

I was trying to use ```BeJsonEquivalentTo``` in some tests, and hit some cases where it said that the json was different, but then the diff output looked to be the same.

Upon testing, If I have a test like this
```fsharp
    [<Fact>]
    let ``Big json should be equivalent`` () =
        let biggerJson = """
            {
                "A": "BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB",
                "X": "YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY"
            }
        """
        biggerJson.Should().BeJsonEquivalentTo(biggerJson)
```
and I debug it, if fails the comparison here, apparently due to different line endings in the subject/expected strings:
![image](https://github.com/user-attachments/assets/550ec8a9-0ef5-4606-9edf-661fde04f4e9)

Fiddling with the settings it looks like 
1) If the json is longer than the ```MaxTotalLineLength``` property in ```FracturedJsonOptions``` then the json is rendered as multi-line using the system EOL (that being ```Crlf``` on Windows)
2) The Regex.Replace at https://github.com/cmeeren/Faqt/blob/7f8011310eb47fad4450039938407b59e45c6661/src/Faqt/StringAssertions.fs#L482 replaces the ```\r\n``` with ```\n``` in the subject string.

and that causes the string compare to fail.

This particular case seems to be fixed by either setting ```MaxTotalLineLength``` to a longer value, or setting ```JsonEolStyle``` in ```FracturedJsonOptions``` to ```FracturedJson.EolStyle.Lf```, though I'm not sure what the best approach is as i'm not entirely sure what the Regex.Replace is for here.

Thanks.

**Comment by cmeeren** [comment]
Thanks for the report! I have fixed this locally and will publish soon together with some other changes.

> The Regex.Replace at … replaces the `\r\n` with `\n` in the subject string.

Actually, I think it replaces `\r\n` with `\r`, which itself is a red flag. In any case, it was a leftover from before I started using `OmitTrailingWhitespace = true` in `FracturedJsonOptions`. When I introduced that, I removed the regex replacement from the expected value, but forgot to remove it from the subject value.

Removing the regex replacement fixed the issue.
